src/downloadextractthread.cpp

```diff
@@ -398,6 +398,48 @@ void DownloadExtractThread::extractMultiFileRun()
                 }
             }
 
+            if(_imgWriterSettings.contains("UNRAID_LANG_JSON")) {
+                // remove this tmp file now that we're definitely done with it
+                QFile langJson(_imgWriterSettings["UNRAID_LANG_JSON"].toByteArray());
+                langJson.remove();
+            }
+
+            if(_imgWriterSettings.contains("UNRAID_LANG_CODE"))
+            {
+                QString unraidLangCode(_imgWriterSettings["UNRAID_LANG_CODE"].toString());
+                if(_imgWriterSettings.contains("UNRAID_LANG_XML")) {
+                    QFile langXml(_imgWriterSettings["UNRAID_LANG_XML"].toByteArray());
+                    QFile::rename(langXml.fileName(), folder + "/config/plugins/lang-" + unraidLangCode + ".xml");
+                }
+
+                if(_imgWriterSettings.contains("UNRAID_LANG_ZIP")) {
+                    QFile langZip(_imgWriterSettings["UNRAID_LANG_ZIP"].toByteArray());
+                    QFile::rename(langZip.fileName(), folder + "/config/plugins/dynamix/lang-" + unraidLangCode + ".zip");
+                }
+                QFile dynamixCfg(folder + "/config/plugins/dynamix/dynamix.cfg");
+                if (dynamixCfg.exists())
+                {
+                    dynamixCfg.open(QIODevice::ReadOnly);
+                    QString dataText = dynamixCfg.readAll();
+                    dynamixCfg.close();
+
+                    QString oldData(dataText);
+                    dataText.replace("locale=\"\"", "locale=\"" + unraidLangCode + "\"", Qt::CaseInsensitive);
+                    if(oldData == dataText)
+                    {
+                        // if this string wasn't found for replacement, just add it
+                        dataText += "locale=\"" + unraidLangCode + "\"";
```

Comment:
We need to make sure we put this locale= block under the "[display]" section - you may want to use an XML parser to parse the file and add a new section, I believe that you may want to parse this as XML

```cpp
#include <QDomDocument>
#include <QString>
#include <QDebug>

void addLocaleToDisplaySection(QString& dataText, const QString& unraidLangCode) {
    QDomDocument doc;
    QString errorMsg;
    int errorLine, errorColumn;

    // Load the dataText as an XML document
    if (!doc.setContent(dataText, &errorMsg, &errorLine, &errorColumn)) {
        qDebug() << "Error parsing XML at line" << errorLine << "column" << errorColumn << ":" << errorMsg;
        return;
    }

    // Find the [display] section
    QDomElement root = doc.documentElement(); // Get the root element
    QDomNodeList displaySections = root.elementsByTagName("display");

    if (!displaySections.isEmpty()) {
        QDomElement display = displaySections.at(0).toElement(); // Assuming only one [display] section

        if (!display.isNull()) {
            // Check if "locale" attribute exists
            if (display.hasAttribute("locale")) {
                display.setAttribute("locale", unraidLangCode);
            } else {
                // Add "locale" attribute if not present
                display.setAttribute("locale", unraidLangCode);
            }
        }
    } else {
        // Add a new [display] section with the locale attribute
        QDomElement newDisplay = doc.createElement("display");
        newDisplay.setAttribute("locale", unraidLangCode);
        root.appendChild(newDisplay);
    }

    // Convert the modified XML back to a string
    QString newXml;
    QTextStream stream(&newXml);
    doc.save(stream, 4); // Indent with 4 spaces
    dataText = newXml;
}
```